packages/dds/tree/src/test/utils.ts

```diff
@@ -144,28 +150,34 @@ export class TestTreeProvider {
      */
     public static async create(
         trees = 0,
-        summarizeOnDemand = false,
+        summarizeType: SummarizeType = SummarizeType.disabled,
         factory: SharedTreeFactory = new SharedTreeFactory(),
     ): Promise<ITestTreeProvider> {
         // The on-demand summarizer shares a container with the first tree, so at least one tree and container must be created right away.
         assert(
-            !(trees === 0 && summarizeOnDemand),
+            !(trees === 0 && summarizeType === SummarizeType.onDemand),
             "trees must be >= 1 to allow summarization on demand",
         );
 
         const registry = [[TestTreeProvider.treeId, factory]] as ChannelFactoryRegistry;
         const driver = new LocalServerTestDriver();
-        const objProvider = new TestObjectProvider(
-            Loader,
-            driver,
-            () =>
-                new TestContainerRuntimeFactory(
-                    "@fluid-example/test-dataStore",
-                    new TestFluidObjectFactory(registry),
-                ),
-        );
+        const containerRuntimeFactory =
+            summarizeType === SummarizeType.disabled
+                ? () =>
+                      new TestContainerRuntimeFactory(
+                          "@fluid-example/test-dataStore",
+                          new TestFluidObjectFactory(registry),
+                          { summaryOptions: { disableSummaries: true } },
```

Comment:
You can put the ternary here to avoid duplicating the call and the other params:
`summarizeType === SummarizeType.disabled ? { summaryOptions: { disableSummaries: true } } : undefined,`

You may even be able to just do this:
`{ summaryOptions: { disableSummaries: summarizeType === SummarizeType.disabled } }`